aspnetcore/fundamentals/websockets.md

```diff
@@ -14,17 +14,50 @@ uid: fundamentals/websockets
 
 This article explains how to get started with WebSockets in ASP.NET Core. [WebSocket](https://wikipedia.org/wiki/WebSocket) ([RFC 6455](https://tools.ietf.org/html/rfc6455)) is a protocol that enables two-way persistent communication channels over TCP connections. It's used in apps that benefit from fast, real-time communication, such as chat, dashboard, and game apps.
 
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/websockets/samples) ([how to download](xref:index#how-to-download-a-sample), [how to run](#sample-app)).
+## Http/2 WebSockets support
+
+Using WebSockets over HTTP/2 takes advantage of new features such as:
+
+* Header compression.
+* Multiplexing, which reduces the time and resources needed when making multiple requests to the server.
+
+These supported features are available in Kestrel on all HTTP/2 enabled platforms. The version negotiation is automatic in browsers and Kestrel, so no new APIs are needed.
+
+.NET 7 introduced Websockets over HTTP/2 support for Kestrel, SignalR JavaScript client and SignalR with Blazor WebAssembly.
+
+> [!NOTE]
+> HTTP/2 WebSockets use CONNECT requests rather than GET, so your own routes and controllers may need updating.
```

Comment:
What needs to be done to them? If they have HttpGet attribute there is no HttpConnect attribute, although this text seems to imply you can just change ...Get to ...Connect.

Also, I tried the Websockets sample app in .NET 7 and it tries to use Http/2 but it doesn't work, so something like what we're telling people to do in their own apps needs to be done in the sample app.

Reply:
What issue were you seeing? Did you use HTTPS?

Reply:
Yes I used HTTPS, I got 405 Method Not Allowed

Reply:
With the `app.Use` sample?

Reply:
Because as you noted in the first paragraph there is no `HttpConnect` attribute (which is what would be needed for controllers).

To fix controllers in the sample we would need something like:
```csharp
public class HttpConnectAttribute : HttpMethodAttribute
{
    private static readonly IEnumerable<string> _supportedMethods = new[] { "CONNECT" };

    public HttpConnectAttribute()
        : base(_supportedMethods)
    {
    }
}
```